src/metabase/query_processor/middleware/cache_backend/db.clj

```diff
@@ -47,7 +44,7 @@
         (throw e)))))
 
 (defn- cached-results [query-hash max-age-seconds respond]
-  (with-open [conn (.getConnection (datasource))
+  (with-open [conn (jdbc/get-connection (db/connection))
```

Comment:
Not specifically related to the fix but my sandboxing test was failing because `toucan.db/connection` was getting bound to `{:connection java.sql.Connection}` by the `with-gtaps` macro which broke this stuff. This change made sense to me anyway since getting a `Connection` from a JDBC spec is something we should let `clojure.java.jdbc` do for us.